File: 0049-group-anagrams/solution.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {

        Map<List<Integer>, List<String>> map = new HashMap<>();

        for (String s : strs) {
            int[] arr = new int[26];

            for (char c : s.toCharArray()) {
                arr[c - 'a']++;
            }

            List<Integer> key = new ArrayList<>();
            for(int i: arr){
                key.add(i);
            }
            
            map.putIfAbsent(key, new ArrayList<String>());
            map.get(key).add(s);
        }

        return new ArrayList<>(map.values());
    }
}

File: 0049-group-anagrams/solution.py
class Solution:
    def groupAnagrams(self, strs: list[str]) -> list[list[str]]:
        mapy = {}

        for s in strs:
            arr = [0] * 26
            
            for char in s:
                index = ord(char) - ord("a")
                arr[index] = arr[index] + 1
            
            lst = mapy.get(str(arr), [])
            lst.append(s)
            mapy[str(arr)] = lst

        return list(mapy.values())
